pkg/cloudprovider/provider/openstack/helper.go

```diff
@@ -302,7 +304,7 @@ func getFreeFloatingIPs(client *gophercloud.ProviderClient, region string, float
 
 	var freeFIPs []osfloatingips.FloatingIP
 	for _, f := range allFIPs {
-		if f.Status == "ACTIVE" && f.PortID == "" {
+		if f.Status != openstackFloatingIPErrorStatusName && f.PortID == "" {
```

Comment:
nit: as you are already in the go package "openstack" theres no need to prefix the constant with "openstack*"
I'll change it upstream